packages/core/src/index.ts

```diff
@@ -0,0 +1,144 @@
+import { handleRequest } from '@graphql-yoga/handler'
+import type { GraphQLSchema } from 'graphql'
+import {
+  Plugin,
+  GetEnvelopedFn,
+  envelop,
+  useMaskedErrors,
+  useExtendContext,
+  enableIf,
+  useLogger,
+  useSchema,
+} from '@envelop/core'
+import { useDisableIntrospection } from '@envelop/disable-introspection'
+import { useValidationCache } from '@envelop/validation-cache'
+import { useParserCache } from '@envelop/parser-cache'
+import { Logger, dummyLogger } from 'ts-log'
+
+/**
+ * Configuration options for the server
+ */
+export type BaseGraphQLServerOptions = {
+  schema: GraphQLSchema
+  /**
+   * Envelop Plugins
+   * @see https://envelop.dev/plugins
+   */
+  plugins?: Array<Plugin>
+  /**
+   * Detect server environment
+   * Default: `false`
+   */
+  isDev?: boolean
+}
+
+/**
+ * Base class that can be extended to create a GraphQL server with any HTTP server framework.
+ */
+export abstract class BaseGraphQLServer {
+  /**
+   * Request handler for helix
+   */
+  protected handleRequest = handleRequest
+  protected schema: GraphQLSchema
+  /**
+   * Instance of envelop
+   */
+  protected envelop: GetEnvelopedFn<any>
+  protected isDev: boolean
+  protected logger: Logger
+
+  constructor(options: BaseGraphQLServerOptions) {
+    this.schema = options.schema
+    this.logger = dummyLogger
+    this.isDev = options.isDev ?? false
+
+    this.envelop = envelop({
+      plugins: [
+        // Use the schema provided by the user
+        useSchema(this.schema),
+        // Performance things
+        useParserCache(),
+        useValidationCache(),
+        // Inject logger instance to context. Useful to use logger in resolvers.
+        useExtendContext(() => ({ logger: this.logger })),
```

Comment:
How can we provide a `GraphQLContext` type that knows about this plugin but as user adds more plugins it automatically update that type?

Reply:
Well I couldn't really get it to work properly yet and even if I extract from it I need to export a type but the envelop instance I need to extract context from is part of a class.